internal/k8shandler/log-exploration-api.go

```diff
@@ -0,0 +1,180 @@
+
+	logApiPodSpec := newLogExplorationApiPodSpec()
+
+	logApiDeployment := NewDeployment("logexplorationapi", clusterRequest.Cluster.Namespace, "logexplorationapi", "logexplorationapi", logApiPodSpec)
+
+	utils.AddOwnerRefToObject(logApiDeployment, utils.AsOwner(clusterRequest.Cluster))
+
+	err := clusterRequest.Create(logApiDeployment)
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+func newLogExplorationApiPodSpec() v1.PodSpec {
+	resources := &v1.ResourceRequirements{
+		Limits: v1.ResourceList{
+			v1.ResourceMemory: defaultLoggingApiMemory,
+			v1.ResourceCPU:    defaultLoggingApiCpuRequest,
+		},
+		Requests: v1.ResourceList{
+			v1.ResourceMemory: defaultLoggingApiMemory,
+			v1.ResourceCPU:    defaultLoggingApiCpuRequest,
+		},
+	}
+
+	logExplorationApiContainer := NewContainer("logexplorationapi", "logApi", v1.PullIfNotPresent, *resources)
+
+	logExplorationApiContainer.Ports = []v1.ContainerPort{
+		{
+			ContainerPort: logAPIPort,
+		},
+	}
+	logExplorationApiContainer.Env = []v1.EnvVar{
+		{Name: "ES_ADDR", Value: "https://elasticsearch.openshift-logging:9200"},
+		{Name: "ES_CERT", Value: "/etc/openshift/elasticsearch/secret/tls.crt"},
+		{Name: "ES_KEY", Value: "/etc/openshift/elasticsearch/secret/tls.key"},
+		{Name: "ES_TLS", Value: "true"},
+		{Name: "POD_IP", ValueFrom: &v1.EnvVarSource{FieldRef: &v1.ObjectFieldSelector{APIVersion: "v1", FieldPath: "status.podIP"}}},
+	}
+
+	logExplorationApiContainer.VolumeMounts = []v1.VolumeMount{
+		{Name: "certificates", MountPath: "/etc/openshift/elasticsearch/secret"},
+	}
+	httpGetStructLivenessProbe := &v1.HTTPGetAction{
+		Path: "/ready",
+		Port: intstr.FromInt(logAPIPort),
+	}
+	httpGetStructReadinessProbe := &v1.HTTPGetAction{
+		Path: "/health",
+		Port: intstr.FromInt(logAPIPort),
+	}
+	handlerStructLivenessProbe := v1.Handler{
+		HTTPGet: httpGetStructLivenessProbe,
+	}
+
+	handlerStructReadinessProbe := v1.Handler{
+		HTTPGet: httpGetStructReadinessProbe,
+	}
+	logExplorationApiContainer.ReadinessProbe = &v1.Probe{
+		Handler:             handlerStructReadinessProbe,
+		InitialDelaySeconds: 3,
+		PeriodSeconds:       3,
+	}
+	logExplorationApiContainer.LivenessProbe = &v1.Probe{
+		Handler:             handlerStructLivenessProbe,
+		InitialDelaySeconds: 10,
+		PeriodSeconds:       3,
+		FailureThreshold:    30,
+	}
+
+	logExplorationApiPodSpec := newLogApiPodSpec([]v1.Container{logExplorationApiContainer},
+		[]v1.Volume{
+			{Name: "certificates", VolumeSource: v1.VolumeSource{Secret: &v1.SecretVolumeSource{SecretName: "fluentd", DefaultMode: utils.GetInt32(420)}}},
+		},
+	)
+	return logExplorationApiPodSpec
+
+}
+
+func newLogApiPodSpec(containers []v1.Container, volumes []v1.Volume) v1.PodSpec {
+
+	return v1.PodSpec{
+		Containers: containers,
+		Volumes:    volumes,
+	}
+}
+
+func (clusterRequest *ClusterLoggingRequest) CreateOrDeleteLogExplorationApi() error {
+	if _, ok := clusterRequest.Cluster.Annotations["api-enabled"]; ok {
```

Comment:
@alanconway @igor-karpukhin  did we settle on enabling this feature via annotation ?  Is this a GA feature?  Can it not just be deployed?  If it's a techpreview feature then I believe the annotation should reflect something about TP in the name